### ISSUE TITLE
Provide more specific messaging when 

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#9148]
## Feature Description

Now that we have the ability to lookup a Tag Manager container by tag ID](https://issuetracker.google.com/issues/330844353), we should implement the more specific messaging for the GTM and GA integration screens that was originally specced for https://github.com/google/site-kit-wp/issues/7990 (see https://github.com/google/site-kit-wp/issues/7989#issuecomment-1915654586).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

- When setting up or editing Tag Manager settings, the messages and screens shown when the GTM container's tag points to a measurement ID should be updated as follows:
  - If GA4 is already connected, having a measurement ID, and the selected Tag Manager container has a Google Tag (`googtag`) with the same tag ID as the measurement ID, then display the following screen (which was originally shown when UA was setup and a property ID matched).
    - Screenshot:
      ![image](https://github.com/google/site-kit-wp/assets/1721319/7d921737-b9cd-419e-8819-6abce9aa313a)
    - The message should be modified to say: "Looks like you’re using Google Analytics. Your Analytics property with measurement ID xxxxxxxxxxx is already set up in your Google Tag Manager configuration, so Site Kit will switch to using Google Tag Manager for Analytics."
    - The CTA should be modified as per the screenshot above (to "Confirm & Continue").
  - If GA4 is not connected, but the selected Tag Manager container has a Google Tag (`googtag`) with a valid measurement ID, then display the following screen:
    - Screenshot:
      ![image](https://github.com/google/site-kit-wp/assets/1721319/53cb2f6c-95ce-425e-a6be-ed8b73ce728b)
    - The message and the CTAs should match the above screenshot.
    - Clicking on "Continue to Analytics setup" should take the user to the Connect Service Analytics screen. (`/wp-admin/admin.php?page=googlesitekit-dashboard&slug=analytics&reAuth=true`). The text which should be added just below the heading and above the "Please select the account information below." line is as follows:
      - When the selected measurement ID in the Analytics form is the same as the measurement ID within the current GTM container's tag: "A Google Tag Manager container with a tag for the selected property with measurement ID G-XXXXXXXXXX, was found on your site, so Site Kit will not place its own tag. If you would prefer to have Site Kit insert this tag, remove it from your Google Tag Manager container and update later in Settings.
      - When the measurement ID is not the same: "An existing Google Analytics property with measurement ID G-XXXXXXXXXX was found on your site, added by Google Tag Manager."

## Implementation Brief

* [ ] 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by jimmymadon** [comment]
@techanvil Thanks for creating this issue. I reviewed this initially and then discussed it on our AC sync today. 

1. As mentioned in [the comment](https://github.com/google/site-kit-wp/issues/7989#issuecomment-1915654586) in issue #7989:

> "We will not check if the Google Tag points to an actual GA4 web data stream until perhaps direct API support is available to find these details from a Google Tag ID"

The above is pretty much the only thing we can do now, i.e. when Analytics is not already connected, we can tell the user for sure now that their Google Tag Manager container contains a tag that points to a Analytics 4 Measurement ID. We agreed that this isn't something that is worth our time and effort as this information doesn't change anything for our users. We already "assume" this to be the case and suggest the user connect Analytics when we implemented issue #7990. 

2. All the remaining cases, which are when Analytics is connected and the Measurement ID in the GTM container matches or does not match the Measurement ID in our Analytics config are pointless now. This is because:
 - UA duplicated events when tags were duplicated on a page. This is why SK was programmed to switch to using the GTM tag and not placing the Analytics tag.
 - Not placing the GA4 tag now in SK has other implications which @aaemnnosttv pointed out - like not tracking custom dimensions, etc. 
 - So we want to make sure that we always insert our own GA4 tag regardless of any tags found within GTM. GA4 does not duplicate events if the same measurement ID tags are duplicated on a page.

So I think we can close this issue. This is good documentation that we have reviewed #7989/#7990 after the Google Tag API development.

c.c. @aaemnnosttv

**Comment by aaemnnosttv** [comment]
Thanks @jimmymadon – it sounds like there isn't much benefit to implementing this as defined as you noted but I'd like to get @techanvil 's perspective in case we've missed anything in our review.

**Comment by techanvil** [comment]
Thanks @jimmymadon @aaemnnosttv.

Sorry, I must have missed the point toward the end of the https://github.com/google/site-kit-wp/issues/7989#issuecomment-1915654586 about the scenarios about switching to Analytics for the tag placement no longer being needed when creating this issue. Makes sense, thanks for clarifying that Jimmy.

> We agreed that this isn't something that is worth our time and effort as this information doesn't change anything for our users.

This also mostly makes sense. It _could_ make a difference in the case where a user doesn't already have a relevant Analytics web data stream set up. We could improve the accuracy of our messaging and the associated user journey, but to fully address this we'd want to ensure that not only does the GTM tag ultimately point to an Analytics measurement ID, but also that the measurement ID is for a web data stream with a URL which matches the current site. That way we could remove the ambiguity from the message ("Looks like you may be using Google Analytics within your Google Tag Manager configuration..."), and ensure the user will always land on a fully populated setup page when clicking through to activate Analytics.

What's debatable is whether it's worth the time and effort to make this change. Personally I think it's out there in the margins but it's still nice to be more accurate, so I do see this as an improvement it would be nice to make as time allows. However a downside would be that it would involve potentially making a couple of additional Google API requests each time we select a new container in the Tag Manager setup flow.

WDYT?

**Comment by jimmymadon** [comment]
@techanvil 

>Personally I think it's out there in the margins but it's still nice to be more accurate, so I do see this as an improvement it would be nice to make as time allows

Yes, this is exactly why I ran this past @aaemnnosttv in our AC sync to just get his thoughts on the ROI here. I remember @marrrmarrr saying that only 10% of Site Kit users had Tag Manager set up - which further reduces the value of this issue and the accurate messaging. So we both thought that it wasn't a "big" enough enhancement that would help many users in any significant way.

>However a downside would be that it would involve potentially making a couple of additional Google API requests each time we select a new container in the Tag Manager setup flow.

Well, this is fine I think as these requests won't have to be made on frequently accessed pages like the dashboard.

So in summary, I could improve the ACs here if we think we want to proceed with this. I would let @aaemnnosttv make the final call here!🙂

**Comment by aaemnnosttv** [comment]
The current setup scenario I think we're discussing here looks like this

<img width="976" height="495" alt="Image" src="https://github.com/user-attachments/assets/100dc2ab-ec0f-4ce9-abe7-3c6c7abe1cae" />

I feel this is good enough for the footprint GTM has in SK because it offers two simple choices – many likely won't read the descriptive text, and if it doesn't change the options we present to choose from, then there is very little value to add.

Perhaps I'm missing some of the context or intent of this old issue, but I'm inclined to close it for the time being. If I'm missing something, please feel free to correct me though.
